### PR TITLE
bump npm & some bower dependencies

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Opening a PR w/an initial commit to save/track comments & todos

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
NPM notes:
- grunt 0.4.5 is highest we can go, jumping to the 1.0.0 generates numerous `npm ERR! peerinvalid Peer` errors

**Comment by spadgett** [comment]
@benjaminapetersen you might look at

https://www.npmjs.com/package/npm-check-updates

**Comment by benjaminapetersen** [comment]
Nice! I'll look at that.  

Also thinking about https://docs.npmjs.com/cli/outdated

**Comment by benjaminapetersen** [comment]
[TEST]

**Comment by benjaminapetersen** [comment]
@jwforres @spadgett ping for review, ill prob do additional bower updates as a separate PR.

File: assets/Gruntfile.js [hunk 1]
```diff
@@ -207,6 +208,8 @@ module.exports = function (grunt) {
           'bower_components/uri.js/src/URI.min.js',
           'bower_components/uri.js/src/jquery.URI.min.js',
           'bower_components/uri.js/src/URI.fragmentQuery.js',
+          // bower registration error? we get 2x versions of uri.js/urijs
+          'bower_components/urijs/',
```
Comment (spadgett) [inline assets/Gruntfile.js]: I'd like to understand why we have two versions. Is this only happening with your changes?

Comment (spadgett) [inline assets/Gruntfile.js]:
```
├─┬ hawtio-core-navigation#2.0.56
│ ├── angular-route#1.3.8 (1.5.6-build.4764+sha.d985944 available)
│ ├── hawtio-core#2.0.11 (2.0.36 available)
│ ├── lodash#3.10.1 (3.11.0-npm-packages available, latest is 4.11.1)
│ └── urijs#1.17.1
```

Comment (spadgett) [inline assets/Gruntfile.js]: Could be a similar problem to "font-awesome" vs "fontawesome"

Comment (benjaminapetersen) [inline assets/Gruntfile.js]: Just confirming, you are seeing it as well?  I believe it appears even when I'm on `console_1.3` branch.  I'll check it again.

**Comment by spadgett** [comment]
Unable to run test-assets with these changes.

```
INFO [karma]: Karma v0.12.37 server started at http://localhost:8443/
WARN [launcher]: Can not load "PhantomJS", it is not registered!
  Perhaps you are missing some plugin?
```

grunt test-integration works, though.

File: assets/Gruntfile.js (hunk 1 above)
Comment (benjaminapetersen) [inline assets/Gruntfile.js]: ping @gashcrumb

**Comment by jwforres** [comment]
@spadgett @benjaminapetersen is this ready to be merged now?

**Comment by benjaminapetersen** [comment]
@jwforres sorry, not quite yet. Tracking down Sam's error I've got a couple more crop up I'm working out.

File: assets/Gruntfile.js (hunk 1 above)
Comment (benjaminapetersen) [inline assets/Gruntfile.js]: Sticking with using `uri.js` & opting out of `urijs`.  Is just messy trying to flip-flop them, really not sure why.  Something must be messed up with the way that repo was double registered (perhaps around the time of the version we are using?), but swapping generates a bunch of errors.

**Comment by benjaminapetersen** [comment]
Runs fine locally after reverting the uri mixups.  Perhaps give it another spin?

File: assets/bower.json [hunk 2]
```diff
@@ -56,6 +56,11 @@
     "font-awesome": "4.6.1"
   },
   "overrides": {
+    "angular" : {
+      "dependencies" : {
+        "jquery": "2.1.4"
```
Comment (jwforres) [inline assets/bower.json]: why an override as opposed to a resolution?

**Comment by jwforres** [comment]
install/test/build all ran clean for me.  just have the one question https://github.com/jwforres/origin/pull/43#discussion_r61319111

File: assets/bower.json (hunk 2 above)
Comment (benjaminapetersen) [inline assets/bower.json]: This is a bit of a "gotcha".  Angular doesn't depend on jQuery, we just use it in place of `$jqLite`.  I had to override to force a dependency else wiredeps will load jQuery after Angular (instead of before), and if that happens it reverts `element` to `$jqLite` (and lots of stuff breaks).

TL;DR
Its not to resolve a version, its because wiredeps needs to think Angular depends on jQuery :smile:

**Comment by benjaminapetersen** [comment]
Cool!  
```
cd assets/
grunt test-integration --suite rest_api 
grunt test-e2e --suite rest_api
```
may be good to run as well

**Comment by benjaminapetersen** [comment]
Closing to open against origin master
